samza-core/src/main/java/org/apache/samza/clustermanager/ClusterBasedJobCoordinator.java

```diff
@@ -230,7 +231,7 @@ public void run() {
       StartpointManager startpointManager = createStartpointManager();
       startpointManager.start();
       try {
-        startpointManager.fanOut(jobModel.getTaskToSystemStreamPartitions());
+        startpointManager.fanOut(new JobModelUtil(jobModel).getTaskToSystemStreamPartitions());
```

Comment:
Consider making this a static helper. If there are any future util methods, then you won't have to build a new object to just use one of them.